Client/Frontend/Home/JumpBackIn/Data/JumpBackInDataAdaptor.swift

```diff
@@ -82,85 +87,85 @@ class JumpBackInDataAdaptorImplementation: JumpBackInDataAdaptor, FeatureFlaggab
         return mostRecentSyncedTab
     }
 
-    // MARK: Jump back in data
-
-    private func updateTabsAndAccountData() {
-        getHasSyncAccount { [weak self] in
-            self?.updateTabsData()
-        }
+    func setDelegate(delegate: JumpBackInDelegate) {
+        self.delegate = delegate
     }
 
-    private func updateTabsData() {
-        updateTabsData { [weak self] in
-            self?.delegate?.didLoadNewData()
-        }
+    // MARK: Jump back in data
 
-        updateRemoteTabs { [weak self] in
-            self?.delegate?.didLoadNewData()
-        }
+    private func updateTabsAndAccountData() async {
+        hasSyncAccount = await getHasSyncAccount()
+        await updateTabsData()
+    }
 
-        if featureFlags.isFeatureEnabled(.tabTrayGroups, checking: .buildAndUser) {
-            updateGroupsData { [weak self] in
-                self?.delegate?.didLoadNewData()
-            }
+    private func updateTabsData() async {
+        recentTabs = await updateRecentTabs()
+        if let remoteTabs = await updateRemoteTabs() {
+            createMostRecentSyncedTab(from: remoteTabs)
         }
+        recentGroups = await updateGroupsData()
```

Comment:
I am writing this just to ensure we cover the case I am mentioning down here. I am unsure this still applies but better be safe than sorry. I tried getting the issue locally but so far nothing on your branch, all seems good to me  🤷‍♀️👍

--
So one of the issues I had in the past with the jbi section, is that each data type can take a different amount of time to load. For instance, we could have local tabs for jump back in, but loading the remote tabs will take a longer amount of time. The section was not updated until we had all data type (so remote tabs loaded), potentially showing outdated local tabs (some could be closed and clicking on them would crash).

To fix this problem, in the past I had `didLoadNewData` called in each data type. So any time "normal tabs" would be loaded, or "remote tabs" would be loaded, the section would update. This made sure we would show up-to-date data in that section, and avoid crashing (from PR https://github.com/mozilla-mobile/firefox-ios/pull/12290, particularly with commit https://github.com/mozilla-mobile/firefox-ios/pull/12290/commits/e1846eee491e3989e7dcd06551aa0069810ff868).